website/docs/language/resources/provisioners/remote-exec.mdx

```diff
@@ -48,8 +48,10 @@ resource "aws_instance" "web" {
 
 The following arguments are supported:
 
-* `inline` - This is a list of command strings. They are executed in the order
-  they are provided. This cannot be provided with `script` or `scripts`.
+* `inline` - This is a list of command strings. Strings are collected in order as
+  a single `.sh` script which is executed on the remote host. A default shell `#!` 
```

Comment:
> as a single `.sh` script

This is a little confusing, it seems to imply the creation of a file where none is created. I think this can safely be omitted. I would change to something such as: "This is a list of command strings to be executed on the remote host. They are executed in the order they are provided."

Reply:
Actually that is the issue which leads to this PR, at least for ssh it **does** create a file on the remote host in order to execute the commands.

I was anticipating that the commands would be executed 1 at a time during an ssh session as implied by the current docs, which would subsequently mean the ssh user's default shell should be used. However, in the remote-exec output I was seeing failures executing "/tmp/somename1234556.sh" because I had was using some bash syntax in my `inline` command strings. When I opened the file it had a `#!/bin/sh` line at the top. I dug through the source code to find that it defaults to this if no `#!` line is found in the script assembled from the inline strings. (somehow I failed to bookmark the code reference, but I can find again if you'd like) 

Once I added an explicit `#!/bin/bash` as the first inline string it resolved my problem.

Reply:
Apologies, I missed this thread! Thanks for the additional context - that makes sense. Perhaps wording such as:

> Strings are collected in order and executed as a single script on the remote host.

As long as this gets past @laurapacilio. :)

Reply:
I'll leave it out since it shouldn't effect anything, so I'm ok to move forward with what we have now.